File: problems/problems_2808/solution.ts
function minimumSeconds(nums: number[]): number {
	const n: number = nums.length;
	const idxMap: Map<number, Array<number>> = new Map();
	for (let i: number = 0; i < n; i++) {
		if (!idxMap.has(nums[i])) {
			idxMap.set(nums[i], []);
		}
		idxMap.get(nums[i]).push(i);
	}
	let ans: number = n;
	// @ts-ignore
	for (const idxes of idxMap.values()) {
		if (idxes.length === 1) {
			continue;
		}
		let cur: number = idxes[0] + n - idxes[idxes.length - 1];
		for (let i: number = 1; i < idxes.length; i++) {
			cur = Math.max(cur, idxes[i] - idxes[i - 1]);
		}
		ans = Math.min(ans, cur);
	}
	return ans >> 1;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(inputValues[0]);
	return minimumSeconds(nums);
}

File: problems/problems_2808/solution.go
package problem2808

import (
	"encoding/json"
	"log"
	"strings"
)

func minimumSeconds(nums []int) int {
	idxMap := map[int][]int{}
	for i, num := range nums {
		idxMap[num] = append(idxMap[num], i)
	}
	n := len(nums)
	ans := n
	for _, idxes := range idxMap {
		cur := idxes[0] + n - idxes[len(idxes)-1]
		for i := 1; i < len(idxes); i++ {
			cur = max(cur, idxes[i]-idxes[i-1])
		}
		ans = min(ans, cur)
	}
	return ans / 2
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}

	return minimumSeconds(nums)
}
